pkg/controller/podgc/gc_controller_test.go

```diff
@@ -54,10 +54,7 @@ func compareStringSetToList(set sets.String, list []string) bool {
 			return false
```

Comment:
Will this be more efficient?

```go
func compareStringSetToList(set sets.String, list []string) bool {
	// 1. Return `false` directly when `len(list) != len(set)`.
	if len(list) != len(set) {
		return false
	}
	// 2. Only when the number of elements of the two are the same,
	// further traversal and judgment are carried out.
	for _, item := range list {
		if !set.Has(item) {
			return false
		}
	}
	return true
 }
```

Reply:
The judgment itself is easy to see, and everyone has his own coding method.
There is no point in discussing this issue. I suggest that we should follow the `lint` specification.

Reply:
This has nothing to do with the coding style you mentioned but has something to do with the efficiency of program execution.

Advance the judgment of whether the element sizes are equal can help improve the execution speed in some cases. 

Assuming such a situation: the elements of the list are **a true subset of the set**, then your code will execute a for range loop in vain.

eg:
```
set: "0", "1", "2", "3", "4", ... , "1000000000"
list:     "1", "2", "3", "4", ... , "1000000000"
```

We can compare them in O(1) instead of O(n)

Reply:
Sorry, I didn’t understand what you meant at first.
Thank you very much, I have modified it,  PTAL.
@BinacsLee